rest/data/admin.go

```diff
@@ -22,18 +24,239 @@ func (ac *DBAdminConnector) GetEvergreenSettings() (*evergreen.Settings, error)
+		if !ok {
+			catcher.Add(errors.New("unable to convert logger config"))
+		}
+		newSettings.LoggerConfig = config
+	}
+	if changes.LogPath != nil {
+		newSettings.LogPath = *changes.LogPath
+	}
+	if changes.NewRelic != nil {
+		i, err := changes.NewRelic.ToService()
+		if err != nil {
+			catcher.Add(err)
+		}
+		config, ok := i.(evergreen.NewRelicConfig)
+		if !ok {
+			catcher.Add(errors.New("unable to convert new relic config"))
+		}
+		newSettings.NewRelic = config
+	}
+	if changes.Notify != nil {
+		i, err := changes.Notify.ToService()
+		if err != nil {
+			catcher.Add(err)
+		}
+		config, ok := i.(evergreen.NotifyConfig)
+		if !ok {
+			catcher.Add(errors.New("unable to convert notify config"))
+		}
+		newSettings.Notify = config
+	}
+	if changes.Plugins != nil {
+		newSettings.Plugins = changes.Plugins
+	}
+	if changes.PprofPort != nil {
+		newSettings.PprofPort = *changes.PprofPort
+	}
+	if changes.Providers != nil {
+		i, err := changes.Providers.ToService()
+		if err != nil {
+			catcher.Add(err)
+		}
+		config, ok := i.(evergreen.CloudProviders)
+		if !ok {
+			catcher.Add(errors.New("unable to convert cloud providers config"))
+		}
+		newSettings.Providers = config
+	}
+	if changes.RepoTracker != nil {
+		i, err := changes.RepoTracker.ToService()
+		if err != nil {
+			catcher.Add(err)
+		}
+		config, ok := i.(evergreen.RepoTrackerConfig)
+		if !ok {
+			catcher.Add(errors.New("unable to convert repotracker config"))
+		}
+		newSettings.RepoTracker = config
+	}
+	if changes.Scheduler != nil {
+		i, err := changes.Scheduler.ToService()
+		if err != nil {
+			catcher.Add(err)
+		}
+		config, ok := i.(evergreen.SchedulerConfig)
+		if !ok {
+			catcher.Add(errors.New("unable to convert scheduler config"))
+		}
+		newSettings.Scheduler = config
+	}
+	if changes.ServiceFlags != nil {
+		i, err := changes.ServiceFlags.ToService()
+		if err != nil {
+			catcher.Add(err)
+		}
+		config, ok := i.(evergreen.ServiceFlags)
+		if !ok {
+			catcher.Add(errors.New("unable to convert service flags config"))
+		}
+		newSettings.ServiceFlags = config
+	}
+	if changes.Slack != nil {
+		i, err := changes.Slack.ToService()
+		if err != nil {
+			catcher.Add(err)
+		}
+		config, ok := i.(evergreen.SlackConfig)
+		if !ok {
+			catcher.Add(errors.New("unable to convert slack config"))
+		}
+		newSettings.Slack = config
+	}
+	if changes.Splunk != nil {
+		i, err := changes.Splunk.ToService()
+		if err != nil {
+			catcher.Add(err)
+		}
+		config, ok := i.(send.SplunkConnectionInfo)
+		if !ok {
+			catcher.Add(errors.New("unable to convert splunk config"))
+		}
+		newSettings.Splunk = config
+	}
+	if changes.SuperUsers != nil {
+		newSettings.SuperUsers = changes.SuperUsers
+	}
+	if changes.Ui != nil {
+		i, err := changes.Ui.ToService()
+		if err != nil {
+			catcher.Add(err)
+		}
+		config, ok := i.(evergreen.UIConfig)
+		if !ok {
+			catcher.Add(errors.New("unable to convert ui config"))
+		}
+		newSettings.Ui = config
+	}
+
+	if persist {
```

Comment:
When would persist be false? What is the purpose of this variable?

Reply:
Setting it to true will save the changes to the DB. False will just apply the changes from the request on top of the current settings. It's used when validating the changes because validation happens on the entire settings document